crates/wasi/src/ctx.rs

```diff
@@ -571,62 +557,6 @@ impl WasiCtxBuilder {
 ///     }
 /// }
 /// ```
-pub trait WasiView: Send {
-    /// Yields mutable access to the internal resource management that this
-    /// context contains.
-    ///
-    /// Embedders can add custom resources to this table as well to give
-    /// resources to wasm as well.
-    fn table(&mut self) -> &mut ResourceTable;
-
-    /// Yields mutable access to the configuration used for this context.
-    ///
-    /// The returned type is created through [`WasiCtxBuilder`].
-    fn ctx(&mut self) -> &mut WasiCtx;
-}
-
-impl<T: ?Sized + WasiView> WasiView for &mut T {
-    fn table(&mut self) -> &mut ResourceTable {
-        T::table(self)
-    }
-    fn ctx(&mut self) -> &mut WasiCtx {
-        T::ctx(self)
-    }
-}
-
-impl<T: ?Sized + WasiView> WasiView for Box<T> {
-    fn table(&mut self) -> &mut ResourceTable {
-        T::table(self)
-    }
-    fn ctx(&mut self) -> &mut WasiCtx {
-        T::ctx(self)
-    }
-}
-
-/// A small newtype wrapper which serves as the basis for implementations of
-/// `Host` WASI traits in this crate.
-///
-/// This type is used as the basis for the implementation of all `Host` traits
-/// generated by `bindgen!` for WASI interfaces. This is used automatically with
-/// [`add_to_linker_sync`](crate::add_to_linker_sync) and
-/// [`add_to_linker_async`](crate::add_to_linker_async).
-///
-/// This type is otherwise provided if you're calling the `add_to_linker`
-/// functions generated by `bindgen!` from the [`bindings`
-/// module](crate::bindings). In this situation you'll want to create a value of
-/// this type in the closures added to a `Linker`.
-#[repr(transparent)]
-pub struct WasiImpl<T>(pub T);
-
-impl<T: WasiView> WasiView for WasiImpl<T> {
-    fn table(&mut self) -> &mut ResourceTable {
-        T::table(&mut self.0)
-    }
-    fn ctx(&mut self) -> &mut WasiCtx {
-        T::ctx(&mut self.0)
-    }
-}
-
 /// Per-[`Store`] state which holds state necessary to implement WASI from this
```

Comment:
These docs are now appended to the ones above, so should these words move above the example perhaps?

Reply:
I made lots of fixes to the docs during the code motion for https://github.com/bytecodealliance/wasmtime/pull/10036 - can we put off the docs review and check it all over there?

Reply:
Sounds reasonable to me 👍